cpp/benchmarks/io/parquet/parquet_writer.cpp

```diff
@@ -16,110 +16,197 @@
 
 #include <benchmarks/common/generate_input.hpp>
 #include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
 #include <benchmarks/io/cuio_common.hpp>
-#include <benchmarks/synchronization/synchronization.hpp>
+#include <benchmarks/io/nvbench_helpers.hpp>
 
 #include <cudf/io/parquet.hpp>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <nvbench/nvbench.cuh>
 
 // to enable, run cmake with -DBUILD_BENCHMARKS=ON
 
 constexpr size_t data_size         = 512 << 20;
 constexpr cudf::size_type num_cols = 64;
 
-namespace cudf_io = cudf::io;
+template <data_type DataType>
+void BM_parq_write_encode(nvbench::state& state, nvbench::type_list<nvbench::enum_type<DataType>>)
+{
+  cudf::rmm_pool_raii rmm_pool;
+
+  auto const data_types             = get_type_or_group(static_cast<int32_t>(DataType));
+  cudf::size_type const cardinality = state.get_int64("cardinality");
+  cudf::size_type const run_length  = state.get_int64("run_length");
+  auto const compression            = cudf::io::compression_type::SNAPPY;
+  auto const sink_type              = io_type::FILEPATH;
```

Comment:
We can speed up these benchmarks if IO is not important.
```suggestion
  auto const sink_type              = io_type::VOID;
```